catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1145,6 +1149,49 @@ public CreateResponse create(CreateRequest createRequest)
         return createResponse;
     }
 
+    private Metacard injectAttributes(Metacard original) {
```

Comment:
Instead of making the CatalogFrameworkImpl larger, please extract the attribute injection concern into another class that can then be used by the CatalogFrameworkImpl.